app/src/main/java/org/wikipedia/readinglist/sync/ReadingListSyncAdapter.kt

```diff
@@ -24,413 +33,418 @@ import org.wikipedia.settings.RemoteConfig
+                readingListSyncNotification.cancelNotification(applicationContext)
+                if (shouldSendSyncEvent) {
+                    SavedPageSyncService.sendSyncEvent(extras.containsKey(SYNC_EXTRAS_REFRESHING))
                 }
-            }
-            if (client.isErrorType(t, "notloggedin")) {
-                try {
-                    L.d("Server doesn't believe we're logged in, so logging in...")
-                    getCsrfToken(wiki, csrfToken)
-                    shouldRetry = true
-                } catch (caught: Throwable) {
-                    errorMsg = caught
-                }
-            }
-            L.w(errorMsg)
-        } finally {
-            Prefs.readingListsLastSyncTime = client.lastDateHeader?.toString() ?: lastSyncTime
-            Prefs.readingListsDeletedIds = listIdsDeleted
-            Prefs.readingListPagesDeletedIds = pageIdsDeleted
-            readingListSyncNotification.cancelNotification(applicationContext)
-            if (shouldSendSyncEvent) {
-                SavedPageSyncService.sendSyncEvent(extras.containsKey(SYNC_EXTRAS_REFRESHING))
-            }
-            if ((shouldRetry || shouldRetryWithForce) && !extras.containsKey(SYNC_EXTRAS_RETRYING)) {
-                val b = Bundle()
-                b.putAll(extras)
-                b.putBoolean(SYNC_EXTRAS_RETRYING, true)
-                if (shouldRetryWithForce) {
-                    b.putBoolean(SYNC_EXTRAS_FORCE_FULL_SYNC, true)
+                if ((shouldRetry || shouldRetryWithForce) && !extras.containsKey(SYNC_EXTRAS_RETRYING)) {
+                    val b = Bundle()
+                    b.putAll(extras)
+                    b.putBoolean(SYNC_EXTRAS_RETRYING, true)
+                    if (shouldRetryWithForce) {
+                        b.putBoolean(SYNC_EXTRAS_FORCE_FULL_SYNC, true)
+                    }
+                    manualSync(b)
                 }
-                manualSync(b)
+                IN_PROGRESS = false
+                SavedPageSyncService.enqueue()
+                L.d("Finished sync of reading lists.")
             }
-            IN_PROGRESS = false
-            SavedPageSyncService.enqueue()
-            L.d("Finished sync of reading lists.")
+            Result.success()
         }
     }
 
-    @Throws(Throwable::class)
-    private fun getCsrfToken(wiki: WikiSite, tokenList: MutableList<String>): String {
-        if (tokenList.size == 0) {
-            tokenList.add(CsrfTokenClient.getToken(wiki).blockingSingle())
+    private fun getBooleanExtraFromData(inputData: Data): Bundle {
+        val extras = Bundle()
+        inputData.keyValueMap.forEach {
+            extras.putBoolean(it.key, it.value as Boolean)
         }
-        return tokenList[0]
+        return extras
+    }
+
+    @Throws(Throwable::class)
+    private suspend fun getCsrfToken(wiki: WikiSite, token: String?): String {
+        return token ?: CsrfTokenClient.getTokenBlocking(wiki)
```

Comment:
Not sure why we were using a list to store tokens but only getting the first one, so I made a change to this function.

Reply:
I think this might have been from a time when there were plans to have _per-wiki_ reading lists, so we would need to sync with each wiki individually. But obviously this is no longer the case.